vmcnet/models/core.py

```diff
@@ -329,7 +329,98 @@ def __call__(self, x: Array) -> Array:  # type: ignore[override]
         return self.final_dense(x)
 
 
-class LogDomainResNet(Module):
+class SimpleSlater(flax.linen.Module):
+    """Slater determinant based on SimpleResNet.
+    Spinful basis functions psi_i can be viewed as
+
+                / phi_up_i(x), s=up
+    psi_i(x,s)={
+               \ phi_down_i(x), s=down
+
+    We define two SimpleResNets phi_list_up/down where
+    phi_list_up(x)=(phi_up_1(x),...,phi_up_{N_up}(x))
+    phi_list_down(x)=(phi_down_{N_up+1}(x),...,phi_down_{N}(x))
+
+    Then we represent the Slater determinant as
+
+    |psi_1(x_1,up)....psi_1(x_{N_up},up),psi_1(x_{N_up+1},down)....psi_1(x_N,down)|
+    |                                                                             |
+    |                                                                             |
+    |                                                                             |
+    |psi_N(x_1,up)....psi_N(x_{N_up},up),psi_N(x_{N_up+1},down)....psi_N(x_N,down)|
+
+    =
+
+    |phi_list_up(x_1)....phi_list_up(x_{N_up}),phi_list_down(x_{N_up+1})....phi_list_down(x_N)|
+
+
+    Attributes:
+        N (int): number of electrons
+        N_up (int): number of spin-up electrons
+        ndense_inner (int): number of dense nodes in layers before the final layer. (SimpleResNet attribute)
+        side_length (int): side length of finite lattice with periodic boundary conditions
+        nlayers (int): number of dense layers applied to the input, including the final
+            layer. If this is 0, the final dense layer will still be applied. (SimpleResNet attribute)
+        activation_fn (Activation): activation function between intermediate layers (is
+            not applied after the final dense layer). Has the signature
+            Array -> Array (shape is preserved) (SimpleResNet attribute)
+    """
+
+    N:int
+    N_up:int
```

Comment:
To be a bit more consistent with the rest of the repo -- could we make this `spin_split`? It would be of type `ParticleSplit`, which is the second argument to `jnp.split`